missing postcss dependency on build

### What version of Next.js are you using?

10.2.0

### What version of Node.js are you using?

v14.15.1

### What browser are you using?

Chrome 90

### What operating system are you using?

macOS 10.15.7

### How are you deploying your application?

next build and start on my local development env

### Describe the Bug

Upgrading from 10.1.1 to 10.2.0 in our package.json and running npm install. 

we get the following dependency error on running `next build` - 

```Failed to compile.
info  - Loaded env from /Users/tso987/Projects/standard-pdp-ui/.env.production
info  - Loaded env from /Users/tso987/Projects/standard-pdp-ui/.env
info  - Using webpack 5. Reason: future.webpack5 option enabled https://nextjs.org/docs/messages/webpack5
next-transpile-modules - WARNING experimental Webpack 5 support enabled
info  - Checking validity of types .next-transpile-modules - WARNING experimental Webpack 5 support enabled
info  - Checking validity of types  
info  - Creating an optimized production build  
Failed to compile.

HookWebpackError: PostCSS plugin postcss-discard-comments requires PostCSS 8.
Migration guide for end-users:
https://github.com/postcss/postcss/wiki/PostCSS-8-for-end-users


> Build error occurred
Error: > Build failed because of webpack errors
    at /Users/tso987/Projects/standard-pdp-ui/node_modules/next/dist/build/index.js:17:924
    at async Span.traceAsyncFn (/Users/tso987/Projects/standard-pdp-ui/node_modules/next/dist/telemetry/trace/trace.js:6:584)
npm ERR! code ELIFECYCLE
```
We aren't using any postcss config in our code.


running `npm ls postcss` shows the missing dependency - 

```
standard-pdp-ui@2.0.0 /Users/tso987/Projects/standard-pdp-ui
├─┬ @JohnLewisPartnership/jl-ui-kit@32.21.0
│ ├─┬ autoprefixer@9.8.6
│ │ └── postcss@7.0.35 
│ ├─┬ css-loader@1.0.1
│ │ ├─┬ icss-utils@2.1.0
│ │ │ └── postcss@6.0.23 
│ │ ├── postcss@6.0.23 
│ │ ├─┬ postcss-modules-extract-imports@1.2.1
│ │ │ └── postcss@6.0.23 
│ │ ├─┬ postcss-modules-local-by-default@1.2.0
│ │ │ └── postcss@6.0.23 
│ │ ├─┬ postcss-modules-scope@1.1.0
│ │ │ └── postcss@6.0.23 
│ │ └─┬ postcss-modules-values@1.3.0
│ │   └── postcss@6.0.23 
│ ├─┬ postcss-inline-svg@3.1.1
│ │ └── postcss@6.0.23 
│ └─┬ postcss-loader@3.0.0
│   └── postcss@7.0.35  deduped
└─┬ next@10.2.0
  └── UNMET PEER DEPENDENCY postcss@8.2.13 
```

I've then had to add postcss as a direct dependency, which then resolves the issue. 

### Expected Behavior

NPM should upgrade next's dependency tree to 10.2, and the app should build correctly

### To Reproduce

Our package.json looks like this - 
```{
  "name": "standard-pdp-ui",
  "description": "Standard product detail page",
  "private": true,
  "version": "v2.0.0",
  "config": {
    "blueprint": {
      "port": 4567,
      "http_probe": "/api/ready",
      "bucket": "standardpdp-assets",
      "hostedAssets": false
    }
  },
  "scripts": {
    "dev": "blueprint-scripts dev",
    "format": "blueprint-scripts format",
    "lint": "blueprint-scripts lint",
    "test": "blueprint-scripts test",
    "cypress": "blueprint-scripts cypress",
    "build": "blueprint-scripts build",
    "start": "blueprint-scripts start",
    "contract": "blueprint-scripts contract",
    "testcafe": "testcafe ${TESTCAFE_BROWSERS:-'chrome:headless'} testcafe/ --skip-js-errors",
    "testcafe-open": "testcafe ${TESTCAFE_BROWSERS:-'chrome'} testcafe/ --skip-js-errors"
  },
  "dependencies": {
    "@JohnLewisPartnership/jl-esi-components": "^6.0.2",
    "@JohnLewisPartnership/jl-ui-kit": "^32.17.0",
    "blueprint": "file:.blueprint",
    "@JohnLewisPartnership/browserslist-config-jl": "^1.4.1",
    "@JohnLewisPartnership/jl-analytics": "0.0.6",
    "@next/bundle-analyzer": "^10.0.9",
    "abort-controller": "^3.0.0",
    "debug": "^4.3.2",
    "deepmerge": "^4.2.2",
    "dotenv": "^8.2.0",
    "dotenv-expand": "^5.1.0",
    "graphql": "^15.5.0",
    "graphql-request": "^3.4.0",
    "identity-obj-proxy": "^3.0.0",
    "isomorphic-unfetch": "^3.1.0",
    "js-cookie": "^2.2.1",
    "newrelic": "^7.1.3",
    "next": "10.2.0",
    "next-compose-plugins": "^2.2.1",
    "next-transpile-modules": "^6.3.0",
    "nookies": "^2.5.2",
    "pino": "^6.11.2",
    "pino-http": "^5.5.0",
    "query-string": "^7.0.0",
    "react": "^17.0.1",
    "react-dom": "^17.0.1",
    "sass": "^1.32.8"
  },
  "devDependencies": {
    "@simplenode/vcr": "^1.3.0",
    "@commitlint/cli": "^11.0.0",
    "@commitlint/config-conventional": "^11.0.0",
    "@google/semantic-release-replace-plugin": "^1.0.2",
    "@pact-foundation/pact": "^9.15.3",
    "@semantic-release/changelog": "^5.0.1",
    "@semantic-release/commit-analyzer": "^8.0.1",
    "@semantic-release/exec": "^5.0.0",
    "@semantic-release/git": "^9.0.0",
    "@semantic-release/gitlab": "^6.0.9",
    "@semantic-release/release-notes-generator": "^9.0.2",
    "@testing-library/dom": "^7.30.0",
    "@testing-library/jest-dom": "^5.11.9",
    "@testing-library/react": "^11.2.5",
    "@testing-library/user-event": "^12.8.3",
    "babel-jest": "^26.6.3",
    "commitlint-config-jira": "^1.5.1",
    "commitlint-plugin-jira-rules": "^1.5.1",
    "cypress": "^6.8.0",
    "eslint": "^7.22.0",
    "eslint-config-prettier": "^8.1.0",
    "eslint-plugin-cypress": "^2.11.2",
    "eslint-plugin-import": "^2.22.1",
    "eslint-plugin-jest-dom": "^3.6.5",
    "eslint-plugin-jsx-a11y": "^6.4.1",
    "eslint-plugin-prettier": "^3.3.1",
    "eslint-plugin-react": "^7.22.0",
    "eslint-plugin-react-hooks": "^4.2.0",
    "eslint-plugin-testing-library": "^3.10.1",
    "husky": "^5.1.3",
    "is-ci": "^3.0.0",
    "jest": "^26.6.3",
    "jest-next-dynamic": "^1.0.1",
    "lint-staged": "^10.5.4",
    "msw": "^0.27.1",
    "npm-merge-driver": "^2.3.6",
    "nyc": "^15.1.0",
    "prettier": "^2.2.1",
    "semantic-release": "^17.4.2",
    "start-server-and-test": "^1.12.1",
    "testcafe": "^1.12.0",
    "testcafe-browser-provider-browserstack": "^1.13.2-alpha.1"
  },
  "keywords": [],
  "author": "",
  "license": "MIT",
  "prettier": "blueprint/config/prettier",
  "eslintConfig": {
    "extends": [
      "./.blueprint/config/eslint"
    ]
  },
  "commitlint": {
    "extends": [
      "./.blueprint/config/commitlint"
    ]
  },
  "release": {
    "extends": "./.blueprint/config/release"
  },
  "husky": {
    "hooks": {
      "commit-msg": "commitlint -E HUSKY_GIT_PARAMS",
      "pre-commit": "blueprint-scripts lint-staged"
    }
  }
}

```

--- a/packages/next/build/webpack/plugins/css-minimizer-plugin.ts
+++ b/packages/next/build/webpack/plugins/css-minimizer-plugin.ts
@@ -48,7 +48,7 @@ export class CssMinimizerPlugin {
       input = asset.source()
     }
 
-    return postcss([cssnanoSimple])
+    return postcss([cssnanoSimple({}, postcss)])
       .process(input, postcssOptions)
       .then((res) => {
         if (res.map) {

--- a/packages/next/build/webpack/plugins/font-stylesheet-gathering-plugin.ts
+++ b/packages/next/build/webpack/plugins/font-stylesheet-gathering-plugin.ts
@@ -18,11 +18,14 @@ import {
 
 function minifyCss(css: string): Promise<string> {
   return postcss([
-    minifier({
-      excludeAll: true,
-      discardComments: true,
-      normalizeWhitespace: { exclude: false },
-    }),
+    minifier(
+      {
+        excludeAll: true,
+        discardComments: true,
+        normalizeWhitespace: { exclude: false },
+      },
+      postcss
+    ),
   ])
     .process(css, { from: undefined })
     .then((res) => res.css)

--- a/packages/next/package.json
+++ b/packages/next/package.json
@@ -79,7 +79,7 @@
     "chokidar": "3.5.1",
     "constants-browserify": "1.0.0",
     "crypto-browserify": "3.12.0",
-    "cssnano-simple": "2.0.0",
+    "cssnano-simple": "2.1.0",
     "domain-browser": "4.19.0",
     "encoding": "0.1.13",
     "etag": "1.8.1",

--- a/packages/next/types/misc.d.ts
+++ b/packages/next/types/misc.d.ts
@@ -3,8 +3,7 @@ declare module 'next/dist/compiled/babel/plugin-transform-modules-commonjs'
 declare module 'next/dist/compiled/babel/plugin-syntax-jsx'
 declare module 'browserslist'
 declare module 'cssnano-simple' {
-  import { OldPlugin } from 'postcss'
-  const cssnanoSimple: OldPlugin<{}>
+  const cssnanoSimple: any
   export = cssnanoSimple
 }
 declare module 'styled-jsx/server'

